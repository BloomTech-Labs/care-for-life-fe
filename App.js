import React from "react";
import Routes from "./src/routes";
import { Provider } from "react-redux";
import store from "./src/store";
import { persistStore } from "redux-persist";
import { PersistGate } from "redux-persist/es/integration/react";

const persistedStore = persistStore(store);

export default function App() {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistedStore} loading={null}>
        <Routes />
      </PersistGate>
    </Provider>
  );
}
