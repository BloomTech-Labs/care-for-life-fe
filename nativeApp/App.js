import "react-native-gesture-handler";
import React, { useState, useEffect } from "react";
import { StyleSheet } from "react-native";
import { ApolloProvider } from "react-apollo";
import { ApolloOfflineProvider } from "react-offix-hooks";
import { offlineClient } from "./config/offix";
import SplashScreen from "./screens/SpashScreen";
import AppNavigator from "./navigation/AppNavigator";
import AuthNavigator from "./navigation/AuthNavigator";

const App = () => {
  const [initialized, setInitialized] = useState(false);
  const [token, setToken] = useState(null);

  // initialize the offix client and set the apollo client
  useEffect(() => {
    offlineClient.init().then(() => setInitialized(true));
  }, []);

  // load the app if the apolloClient is there, otherwise load the splash screen
  if (initialized) {
    return (
      <ApolloOfflineProvider client={offlineClient}>
        <ApolloProvider client={offlineClient}>
          {token === null ? (
            <AuthNavigator />
          ) : (
            <AppNavigator style={styles.container} />
          )}
        </ApolloProvider>
      </ApolloOfflineProvider>
    );
  }
  return <SplashScreen />;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default App;
