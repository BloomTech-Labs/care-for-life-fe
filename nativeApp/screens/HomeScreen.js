import React from "react";
import { StyleSheet, View, Text } from "react-native";

const HomeScreen = (props) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Annual Survey</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontWeight: "bold",
    fontSize: 24,
  },
});

export default HomeScreen;
