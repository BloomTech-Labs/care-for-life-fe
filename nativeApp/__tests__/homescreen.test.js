import React from 'react';
import App from '../App.js';
import renderer from 'react-test-renderer';
import {styles} from '../screens/HomeScreen.js'

import {
    Text,
    TextInput,
    StyleSheet
  } from 'react-native';

test('Homescreen.js Tests Running Correctly', () => {
  expect(1).toBe(1);
});

test('Renders Text Correctly', () => {
  const tree = renderer.create(<Text value="Annual Survey" style={styles.text}/>).toJSON();
  expect(tree).toMatchSnapshot();
});