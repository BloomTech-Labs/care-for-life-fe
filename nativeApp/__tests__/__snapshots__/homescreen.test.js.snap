// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Text Correctly 1`] = `
<Text
  style={
    Object {
      "fontSize": 24,
      "fontWeight": "bold",
    }
  }
  value="Annual Survey"
/>
`;
